c++/src/capnp/message.h

```diff
@@ -195,6 +199,12 @@ class MessageBuilder {
   void setRoot(Reader&& value);
   // Set the root struct to a deep copy of the given struct.
 
+  template <typename Reader>
+  void canonicalRoot(Reader&& value);
```

Comment:
Perhaps the right interface to canonicalization is more like:

    template <typename Reader>
    kj::Array<word> canonicalize(Reader reader);

This would:

1. Call `reader.totalSize()` to get an upper bound on the canonical size.
2. Allocate a `FlatMessageBuilder`.
3. Call `getRoot<AnyPointer>().setCanonical()` to copy and canonicalize (no need for a new method of `MessageBuilder`).
4. Copy out the allocated part of the first segment and return it.